pkg/apis/events/v1beta1/cloudstoragesource_conversion_test.go

```diff
@@ -32,3 +61,50 @@ func TestCloudStorageSourceConversionBadType(t *testing.T) {
 		t.Errorf("ConvertFrom() = %#v, wanted error", good)
 	}
 }
+
+func TestCloudStorageSourceConversion(t *testing.T) {
+	// Just one for now, just adding the for loop for ease of future changes.
+	versions := []apis.Convertible{&v1.CloudStorageSource{}}
+
+	tests := []struct {
+		name string
+		in   *CloudStorageSource
+	}{{
+		name: "min configuration",
+		in: &CloudStorageSource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:       "ps-name",
+				Namespace:  "ps-ns",
+				Generation: 17,
+			},
+			Spec: CloudStorageSourceSpec{},
+		},
+	}, {
+		name: "full configuration",
+		in:   completeCloudStorageSource,
+	}}
+	for _, test := range tests {
+		for _, version := range versions {
+			t.Run(test.name, func(t *testing.T) {
+				ver := version
+				// DeepCopy because we will edit it below.
+				in := test.in.DeepCopy()
+				if err := in.ConvertTo(context.Background(), ver); err != nil {
+					t.Errorf("ConvertTo() = %v", err)
+				}
+				got := &CloudStorageSource{}
+				if err := got.ConvertFrom(context.Background(), ver); err != nil {
+					t.Errorf("ConvertFrom() = %v", err)
+				}
+				// ServiceAccountName and PayloadFormat only exists in v1alpha1 and v1beta1, they doesn't exist in v1.
+				// So it won't round trip, it will be silently removed.
+				in.Status.ServiceAccountName = ""
+				in.Spec.PayloadFormat = ""
```

Comment:
Is there a need to copy the object? Since the test.in will not be used after doing the ConvertTo, I think we can directly modify the fields here.

Reply:
Just want to double check, I can fix them in my PR.

Reply:
`in` is a pointer. If you don't deep copy it and directly modify it, it will affect other files who are using the same pointer.

Reply:
Yes we should keep it.